perf-tool/src/monitor.cpp

```diff
@@ -55,7 +57,19 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
         sprintf(str, "0x%x", hash);
         j["monitorHash"] = str;
     }
-
+   
+    /*monitor for protecting the hashtable*/
+    std::mutex monitorMapMutex;
+    int val = 1;
+    {
+    std::lock_guard<std::mutex> lg(monitorMapMutex);	    
+    if(m_monitor.find(hash) == m_monitor.end())
+        m_monitor[hash] = val;
+    else
+        val = ++(m_monitor[hash]);    
+    }
+    j["per_object_contended_count"] = val;
+    
```

Comment:
I was suggesting a simpler code which avoids repeated searching in the hashtable
```
                    std::lock_guard<std::mutex> lg(monitorMapMutex);
                    val = ++(m_monitor[hash]);
```